### DISCUSSION TITLE
Median PET values

### DISCUSSION THREAD

**Discussion by lennart81** [ENHANCE-PET/MOOSE#140] [General]
I used the cardiac segmentation today and the median PET values are always identical for 9 out of 10 segments. That's odd:
![image](https://github.com/user-attachments/assets/18f9d864-9bbf-4cd8-8d2b-f66bdfefa4eb)

**Comment by LalithShiyam** [comment]
Yep that's odd. @mprires and @Keyn34, I am outside, can you check this out please?

**Comment by LalithShiyam** [comment]
So checked the code @lennart81... seems alright. Can you share the data? The one that caused the issue?
-Lalith

**Comment by lennart81** [comment]
Hey Lalith,

I ran the cardiac segmentation with 6 PET/CT from 2 sites (you just visited one of them according to LinkedIn). I can send you all output files from moose except the patient images.

Would that help?

Best, Jan

**Reply by LalithShiyam** [reply]
Nope. I need to see where it's going wrong. Pet image, ct image and the label would help. If it's caused by the MEL site, I can test it on one of their data.

**Comment by lennart81** [comment]
it is also happening with data from the MEL site. LMK if you need further information

**Reply by LalithShiyam** [reply]
Then I will just check one of the data from MEL site.

**Comment by LalithShiyam** [comment]
We get the same error as well. Constant medians...@lennart81. We looked at the code and it looks alright. The rest of the values are alright, we compared it with slicer and itksnap. Can be one of the two issues: either simpleItk does it wrong or we have made a hypernuanced bug. Either way this would take time.

**Comment by Keyn34** [comment]
@lennart81 We have found the "issue". It has to do with the binning of the `SimpleITK` filter that is used. I will try to find a way to determine an appropriate binning size dynamically and update the repository tomorrow.

**Comment by LalithShiyam** [comment]
@Keyn34 thanks for finding/fixing the binning issue. I pushed `moosez==2.6.4` to pypi and this should fix it. @lennart81 please upgrade `moosez` and let us know, if this is solved.
